### PR TITLE
Alert component

### LINKED ISSUE #601: Core Developer: Alert component
Build a component for alert messages.

``` gherkin
Feature: Contextual Feedback
  As a Core Developer,
  I want to provide users with a feedback message that requires their attention 
  So that users understand if an action is needed and what they need to do

  Scenario: Alerts
    Given that, a user may require contextual feedback
    When they perform an action in Punchcard that required feedback
    Then they will see an alert message
      And the message will be displayed in context of the related element 
      And the message will state the issue and how the user can take action
```
---

**Contextual Feedback:**
- [ ] Alerts

---
Related to #225

### PR DESCRIPTION
**Pull request by ineedsubstance**
Created a component for alerts.

---
Resolves #601 

`DCO 1.1 Signed-off-by: NICK TILDEN <NICK@INEEDSUBSTANCE.COM>`

### CODE REVIEW

File: views/components/alert.html [hunk 1]
```diff
@@ -1,3 +1,5 @@
-{% macro alert(for, type, value) %}
-<p class="alert--{{type | upper}}" role="alert" for="{{for}}">{{value}}</p>
+{% macro alert(data) %}
+{% for item in data %}
+<p class="form--alert" role="alert" for="{{item.for}}"><strong class="form--alert-type">{{item.type}} </strong>{{item.value}}</p>
```
Comment (scottnath) [inline views/components/alert.html]: What criteria determines when `strong` is used?

File: lib/routes/data/styles.js [hunk 2]
```diff
@@ -63,4 +63,16 @@ module.exports = {
       ],
     ],
   },
+  alertData: [
+    {
+      for: '00eef724-7cbd-47f5-a735-630510f39711',
+      type: 'Warning!',
```
Comment (scottnath) [inline lib/routes/data/styles.js]: The alert for forms does not currently have two options. It is just a single message. I think this data should just be `for` and `value`.

File: views/components/alert.html (hunk 1 above)
Comment (scottnath) [inline views/components/alert.html]: if we remove `type` from the alert data, can we then get rid of `strong`?

File: views/components/alert.html [hunk 3]
```diff
@@ -1,3 +1,5 @@
-{% macro alert(for, type, value) %}
-<p class="alert--{{type | upper}}" role="alert" for="{{for}}">{{value}}</p>
+{% macro alert(data) %}
+{% for item in data %}
```
Comment (scottnath) [inline views/components/alert.html]: This component shouldn't check for an array. It should take a single object and create a single alert. If we want multiple components in our styles view, then we should just call the component twice.

Comment (ineedsubstance) [inline views/components/alert.html]: ok, makes senes.

File: views/components/alert.html (hunk 1 above)
Comment (ineedsubstance) [inline views/components/alert.html]: Currently, all alerts look the same and have the same level of visual importance. I was attempting to add some distinction between different alerts by adding the `type`. I will remove for now and create an issue to explore this idea in the future.

screen shot 2016-12-08 at 3 52 53 pm

Comment (ineedsubstance) [inline views/components/alert.html]: yes, this is related to you comment above.

File: lib/routes/data/styles.js (hunk 2 above)
Comment (ineedsubstance) [inline lib/routes/data/styles.js]: yes, this is related to you comment above.

**Comment by ineedsubstance** [comment]
@scottnath - changes have been addressed.

File: views/components/alert.html (hunk 3 above)
Comment (scottnath) [inline views/components/alert.html]: Still checking for an an array, should not be.

File: lib/routes/data/styles.js [hunk 4]
```diff
@@ -63,4 +63,10 @@ module.exports = {
       ],
     ],
   },
+  alertData: [
```
Comment (scottnath) [inline lib/routes/data/styles.js]: should not be an array, or if it is, each object should be called separately below.

File: views/styles.html [hunk 5]
```diff
@@ -329,6 +329,12 @@ <h1 class="base--h2">Form elements</h1>
   </section>
 
   <section class="_style-guide--group">
+    <h2 class="_style-guide--title">Alerts</h2>
+    {% from "components/alert.html" import alert %}
+    {{alert(alertData)}}
```
Comment (scottnath) [inline views/styles.html]: if you're going to have the data be an array for this, you need your `for` statement here, not in the component

File: views/components/alert.html (hunk 3 above)
Comment (ineedsubstance) [inline views/components/alert.html]: ah, my bad. will fix.

File: lib/routes/data/styles.js (hunk 4 above)
Comment (ineedsubstance) [inline lib/routes/data/styles.js]: yep, got lazy. fixing

File: views/styles.html (hunk 5 above)
Comment (ineedsubstance) [inline views/styles.html]: nope, fixed above

**Pull request merged**